src/hooks/use-authentication.ts

```diff
@@ -6,18 +6,30 @@ import { SessionData, UserData, ValidAuthProviderId } from 'types/auth'
  * handle invoking the wrapped function with a default value.
  */
 const signInWrapper = (
-	provider: ValueOf<typeof ValidAuthProviderId> = ValidAuthProviderId.CloudIdp
+	provider: ValidAuthProviderId = ValidAuthProviderId.CloudIdp
 ) => {
 	return signIn(provider)
 }
 
+interface UseAuthenticationOptions {
+	/**
+	 * If `true`, next-auth will automatically redirect to the sign-in page
+	 */
+	isRequired?: boolean
+}
+
 /**
  * Hook for consuming user, session, and authentication state. Sources all data
  * from next-auth/react's `useSession` hook.
  */
-const useAuthentication = () => {
-	// Pull data and status from next-auth's hook
-	const { data, status } = useSession()
+const useAuthentication = (options: UseAuthenticationOptions = {}) => {
+	// Get option properties from `options` parameter
+	const { isRequired = false } = options
+
+	// Pull data and status from next-auth's hook, and pass options
+	const { data, status } = useSession({
+		required: isRequired,
```

Comment:
blocker: Nice find, and I like its intent! However, in our case we do not want to send folks to the interstitial login page with the list of providers:

![Screen Shot 2022-07-22 at 9 10 51 AM](https://user-images.githubusercontent.com/1289701/180457275-b9c65e3b-df39-4c24-9563-562826affe93.png)
 
We want to route directly to the HCP login page, which is why we have the default `cloud-idp` argument for our `signIn` wrapper.

We could potentially implement our own `isRequired` functionality here based on `next-auth`: https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/react/index.tsx#L100-L109

Reply:
@BRKalow Updated! I added an `onUnauthenticated` option to the `useAuthentication` hook, and set up a default for the callback. What do you think?

https://github.com/hashicorp/dev-portal/pull/729/commits/d072fc0827a395b9e9ea7e5158f393fe3715906a